### ISSUE TITLE
Providing an AppImage version of CorsixTH

### ISSUE THREAD

**Issue by XploitOcelot** [CorsixTH/CorsixTH#2726]
### What feature or enhancement would you like to add?

It would be really nice from you guys to start providing us, Linux basic users, with an AppImage version of CorsixTH, so we don't depend anymore on outdated repositories (as you can see down below):

https://repology.org/project/corsixth/versions

I'm a EndeavourOS user, so the latest available version for me is 0.67 (AUR)

### How would this feature benefit the gameplay of CorsixTH?

Linux basic users could try the lastest version of CorsixTH and join its community without needing to spend (a lot of) time learning how to compile packages for any particular distro.

They would just need to enjoy the latest fruits of its (your) development.

### Was the feature in the original game?

No

### What version of CorsixTH are you using?

0.67

### Does this feature/enhancement relate to a current problem?

No

### Additional Information

https://appimage.org/

**Comment by lewri** [comment]
Thanks for reaching out. Outside our current release packages; we usually have external packagers maintain other formats.

Would you be interested in creating an maintaining an appimage for us?

**Comment by XploitOcelot** [comment]
> Thanks for reaching out. Outside our current release packages; we usually have external packagers maintain other formats.
> 
> Would you be interested in creating an maintaining an appimage for us?

I lack the knowledge to compile the CorsixTH package even for myself. How could I help with the AppImage version?? 😬

**Comment by lewri** [comment]
It's not an issue 😄 just good to ask in case. I've marked this with Help Wanted for now as I'm not sure if other devs here are able to task this on.

**Comment by Alberth289346** [comment]
According to Wikipedia, EndavourOS is derived from Arch Linux.
ArchLinux has two corsix packages:
- https://aur.archlinux.org/packages/corsix-th (last update in 2023, seemingly 0.67).
- https://aur.archlinux.org/packages/corsix-th-git (last update in 2020, seemingly a git revision in 0.64)

**Comment by lewri** [comment]
Yeah, I think the maintainer for the AUR has stopped too. Arch likely needs a new maintainer.

**Comment by TheCycoONE** [comment]
https://flatpak.org/setup/EndeavourOS is probably your best bet for getting CorsixTH 0.68 on Endeavour at the moment.

**Comment by lewri** [comment]
Aur has been picked up again. I believe this was updated @envolution  going by their aur fork commits?

**Comment by envolution** [comment]
@lewri yes 0.68.0 is available on aur https://aur.archlinux.org/packages/corsix-th - not sure why the package was originally named the way it was, I've just been keeping it up to date

**Comment by lewri** [comment]
> > Thanks for reaching out. Outside our current release packages; we usually have external packagers maintain other formats.
> > Would you be interested in creating an maintaining an appimage for us?
> 
> I lack the knowledge to compile the CorsixTH package even for myself. How could I help with the AppImage version?? 😬

I am having a poke around at this. On my Ubuntu based VM I've hit an error with `appimage-builder`. I'm going to try on my other laptop running Manjaro to see if it'd build there.

**Comment by lewri** [comment]
An AppImage has been built, awaiting testing. It's currently with beta testers on our Discord server.

**Comment by TheCycoONE** [comment]
Are you going to keep packaging it every release lewri?

**Comment by lewri** [comment]
> Are you going to keep packaging it every release lewri?

If Colin and myself make sure everything is ironed out it can be done for release yes. We can then add 0.68.0 to the releases post as well if you like?
AppImage wants the package to be compatible with the oldest Ubuntu LTS in main support. This is currently 20.04. For 0.69.0 it will be 22.04. But that's more if we want to submit it into their catalogue.

**Comment by TheCycoONE** [comment]
Ok. Is that going to be a problem with sdl3 - assuming it's otherwise ready.

**Comment by legluondunet** [comment]
Hello Corsix-Th team, just tested your AppImage and it doesn't launch on Manjaro:

```
$ ./CorsixTH-0.68.0-x86_64.AppImage 


---------------------------------------------------------------

Welcome to CorsixTH v0.68.0!

---------------------------------------------------------------

This window will display useful information if an error occurs.

---------------------------------------------------------------

An error has occurred in CorsixTH:
...t_CorsixnIIFFN/usr/bin/../share/corsix-th/Lua/strict.lua:66: module 'lpeg' not found:
	no field package.preload['lpeg']
	no file '/tmp/.mount_CorsixnIIFFN/usr/bin/../share/corsix-th/Lua/lpeg.lua'
	no file '/tmp/.mount_CorsixnIIFFN/usr/bin/../share/corsix-th/Lua/lpeg/init.lua'
	no file '/usr/local/share/lua/5.4/lpeg.lua'
	no file '/usr/local/share/lua/5.4/lpeg/init.lua'
	no file '/usr/local/lib/lua/5.4/lpeg.lua'
	no file '/usr/local/lib/lua/5.4/lpeg/init.lua'
	no file '/usr/share/lua/5.4/lpeg.lua'
	no file '/usr/share/lua/5.4/lpeg/init.lua'
	no file './lpeg.lua'
	no file './lpeg/init.lua'
	no file '/tmp/.mount_CorsixnIIFFN/usr/bin/../share/corsix-th/lpeg.so'
	no file '/usr/local/lib/lua/5.4/lpeg.so'
	no file '/usr/lib/x86_64-linux-gnu/lua/5.4/lpeg.so'
	no file '/usr/lib/lua/5.4/lpeg.so'
	no file '/usr/local/lib/lua/5.4/loadall.so'
	no file './lpeg.so'
stack traceback:
	[C]: in upvalue 'fn'
	...t_CorsixnIIFFN/usr/bin/../share/corsix-th/Lua/strict.lua:66: in function 'require'
	...FFN/usr/bin/../share/corsix-th/Lua/string_extensions.lua:22: in main chunk
	[C]: in function 'persist.dofile'
	...unt_CorsixnIIFFN/usr/bin/../share/corsix-th/CorsixTH.lua:70: in field 'require'
	...ount_CorsixnIIFFN/usr/bin/../share/corsix-th/Lua/app.lua:265: in method 'init'
	...unt_CorsixnIIFFN/usr/bin/../share/corsix-th/CorsixTH.lua:116: in main chunk
	[C]: in ?
```

As you can see, it misses several dependencies.

My config:

My config:
Manjaro 25.0.0 Zetar
Kernel: x86_64 Linux 6.12.20-2-MANJARO
Proc: AMD Ryzen 9 5900HX
GPU: AMD Radeon RX 6600M
Drivers: Mesa 25.0.3
Resolution: 1920x1080
32 Go RAM
DE: Xfce4

**Comment by legluondunet** [comment]
I just tested today's AppImage, it still missing some dependencies:

lfs.so
lpeg.so

So, on Ubuntu packages search, I downloaded this two packages:
https://packages.ubuntu.com/noble/amd64/lua-lpeg/download
and
https://packages.ubuntu.com/noble/amd64/lua-lpeg/download
then copy 
liblua5.4-filesystem.so.0.0.0 to lfs.so
and
liblua5.4-lpeg.so.2.0.0 to lpeg.so
I put lfs.so  and lpeg.so to the same folder as the CorsixTH AppImage and it finally launched.

What are you using to make AppImage? Your tool should put all necessary libraries inside AppImage, do you know why not this two librairies?
And what will be your low end supported Ubuntu distribution: Ubuntu 22.04 or 24.04?

**Comment by TheCycoONE** [comment]
Probably because those libraries aren't linked in any way to the code that the compiler/tooling knows about, unless you static compile them in which we have an option for. The lua modules just follow a lua convention and work more like plug-ins. Important to add them.

**Comment by colinjmatt** [comment]
I'm going to actively work on this today for a couple of hours. I have a "working" AppImage that doesn't seem to complain about missing lpeg/lfs.so files, but instead it complains of other missing modules that it's expecting as part of the "Linux distort" it's built with.

I'm going to test the AppImage with a few different LiveCDs and post the fruit of my output sometime soon.

**Comment by colinjmatt** [comment]
New AppImage live to test in our Discord. Tested on:
- Ubuntu 25.04 LiveCD
- Debian 12 fresh install
- Steam OS
- Fedora 42 LiveCD
- Mint 22.1 LiveCD
- MX 23.6 LiveCD
- Manjaro 25.0.1 LiveCD

**Comment by colinjmatt** [comment]
A small consensus suggests that the AppImage now works as it should. I feel we can wait a couple of weeks and close this down as resolved if there's no issues.

We can reliably make these on-demand now.

**Comment by XploitOcelot** [comment]
Hi! Wanted to give a try to the AppImage version, but I've been looking for the file to test it. No luck on finding it. Where is it?

**Comment by colinjmatt** [comment]
@XploitOcelot I can add you into the channel if you want to test?

**Comment by XploitOcelot** [comment]
> [@XploitOcelot](https://github.com/XploitOcelot) I can add you into the channel if you want to test?

Sure! I could give some extensive testing on EndeavourOS. 
It should be a good indicator of the performance of the appimage version on any Arch-based distro installation.

**Comment by lewri** [comment]
An AppImage will now be provided, hosted in our releases page, starting from 0.69.0 (for those who tested 0.68.0 thank you very much).

**Comment by lewri** [comment]
AppImage/appimage.github.io#3492
